src/zypper_plugins/tracer_upload.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python
+#
+# Copyright 2019 ATIX AG
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+
+import json
+import httplib
+from os import path, environ
+import sys
+import logging
+from rhsm.config import RhsmConfigParser, initConfig
+from zypp_plugin import Plugin
+
+sys.path.append('/usr/share/rhsm')
+from subscription_manager.identity import ConsumerIdentity
+
+REBOOT_NEEDED_FLAG = "/var/run/reboot-needed"
+
+class TracerUploadPlugin(Plugin):
+    def __init__(self):
+        Plugin.__init__(self)
+        self.num1 = self.num2 = None
+        self.description = "Upload traces to katello"
+        self.cleanup = ""
+        self.userdat = {}
+
+    def collect_data(self):
+        """
+        Return a array with nested arrays
+        containing name, how to restart & app type
+        for every package that needs restarting
+        """
+
+        apps = {}
+
+        if path.isfile(REBOOT_NEEDED_FLAG)
+            apps["kernel"] = { "helper": "You will have to reboot your computer", "type": "static" }
+        return apps
+
+    def upload_tracer_profile(self):
```

Comment:
We have some code that attempts to de-duplicate functions like `upload_tracer_profile` and I think it might work here:

https://github.com/Katello/katello-host-tools/blob/master/src/katello/tracer.py

Reply:
Well, the functionality on SUSE is pretty different than on EL.

Reply:
Indeed! Your code works just fine but I would prefer something like this:https://github.com/jturel/katello-host-tools/commit/f63c70a4b2af9d9b54d31b691071b233666c10a5

It keep the internals out of the plugin. Mind using this approach and adding some basic tests? I think we have some to model after.